src/worldcereal/job.py

```diff
@@ -88,6 +152,8 @@ def generate_map(
     backend_context: BackendContext,
     output_path: Optional[Union[Path, str]],
     product_type: WorldCerealProduct = WorldCerealProduct.CROPLAND,
+    cropland_parameters: CropLandParameters = CropLandParameters(),
+    croptype_parameters: CropTypeParameters = CropTypeParameters(),
```

Comment:
it's not confusing to a user that even if `product_type=='cropland'`, there is `croptype_parameters` given?

Reply:
Indeed, I thought about it but I didn't find a trivial solution for this.

Maybe one solution would be to change the type from
`croptype_parameters: CropTypeParameters = CropTypeParameters()`
to 
`croptype_parameters: Optional[CropTypeParameters] = CropTypeParameters()`

In terms of logic this would change nothing practically but at least it is more clear for the user. I don't want to set it to `None` by default because that would throw off users that try to run croptype mapping without digging in the parameter's code (I assume most users want to use the default parameters when starting using our package).

I can also document this aspect in the docstring of the `generate_map` function

Reply:
OK let's go for that for now